table.go

```diff
@@ -47,84 +59,71 @@ type RoutingTable struct {
 // NewRoutingTable creates a new routing table with a given bucketsize, local ID, and latency tolerance.
 func NewRoutingTable(bucketsize int, localID ID, latency time.Duration, m peerstore.Metrics) *RoutingTable {
 	rt := &RoutingTable{
-		Buckets:     []*Bucket{newBucket()},
-		bucketsize:  bucketsize,
-		local:       localID,
-		maxLatency:  latency,
-		metrics:     m,
-		PeerRemoved: func(peer.ID) {},
-		PeerAdded:   func(peer.ID) {},
+		Buckets:        []*Bucket{newBucket()},
+		bucketsize:     bucketsize,
+		local:          localID,
+		maxLatency:     latency,
+		metrics:        m,
+		cplRefreshedAt: make(map[uint]time.Time),
+		PeerRemoved:    func(peer.ID) {},
+		PeerAdded:      func(peer.ID) {},
 	}
 
 	return rt
 }
 
-// GetAllBuckets is safe to call as rt.Buckets is append-only
-// caller SHOULD NOT modify the returned slice
-func (rt *RoutingTable) GetAllBuckets() []*Bucket {
-	rt.tabLock.RLock()
-	defer rt.tabLock.RUnlock()
-	return rt.Buckets
-}
+// GetTrackedCplsForRefresh returns the Cpl's we are tracking for refresh.
+// Caller is free to modify the returned slice as it is a defensive copy.
+func (rt *RoutingTable) GetTrackedCplsForRefresh() []*CplRefresh {
```

Comment:
nit: we can avoid a bunch of allocations by removing the pointer. That is, `[]CplRefresh`.

Reply:
Done.